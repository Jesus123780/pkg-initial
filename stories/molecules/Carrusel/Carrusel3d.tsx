import PropTypes from 'prop-types'
import React from 'react'
import styles from './styles.module.css'

export const Carrusel3D = (props) => {
  const {
    children,
    active,
    moveRight,
    moveLeft,
    maxView
  } = props
  const MAX_VISIBILITY = maxView ?? 3
  const count = React.Children.count(children)
  return (
    <>
      <div className={styles['carousel']}>
        {React.Children.map(children, (child, i) => {return (
          <div
            className={styles['card-container']}
            style={{
              '--active': i === active ? 1 : 0,
              '--offset': (active - i) / 3,
              '--direction': Math.sign(active - i),
              '--abs-offset': Math.abs(active - i) / 3,
              pointerEvents: active === i ? 'auto' : 'none',
              opacity: Math.abs(active - i) >= MAX_VISIBILITY ? '0' : '1',
              display: Math.abs(active - i) > MAX_VISIBILITY ? 'none' : 'block'
            }}
          >
            {child}
          </div>
        )})}
      </div>
      <div>
        <div>
          <button disabled={active === 0} onClick={moveLeft}>
            Click
          </button>
        </div>

        <div>
          <button
            disabled={active === count - 1}
            onClick={moveRight}
          >
            Click
          </button>
        </div>
      </div>
    </>
  )
}

Carrusel3D.propTypes = {
  active: PropTypes.number,
  children: PropTypes.any,
  maxView: PropTypes.number,
  moveLeft: PropTypes.any,
  moveRight: PropTypes.any
}
