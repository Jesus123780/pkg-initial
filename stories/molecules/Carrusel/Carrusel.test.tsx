import { render } from '@testing-library/react'
import React from 'react'
import { Carrusel3D } from './Carrusel3d'

describe('Carrusel3D', () => {
  it('renders correctly', () => {
    const { container } = render(<Carrusel3D />)
    expect(container).toMatchSnapshot()
  })
})
