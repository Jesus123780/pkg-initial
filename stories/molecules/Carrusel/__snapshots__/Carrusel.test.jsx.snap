// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrusel3D renders correctly 1`] = `
<div>
  <div />
  <div>
    <div>
      <button>
        Click
      </button>
    </div>
    <div>
      <button>
        Click
      </button>
    </div>
  </div>
</div>
`;
